core/src/commonMain/kotlin/com/xebia/functional/xef/llm/Chat.kt

```diff
@@ -46,35 +46,23 @@ interface Chat : LLM {
   ): Flow<String> = flow {
     val memories: List<Memory> = memories(conversationId, context, promptConfiguration)
 
-    val promptWithContext: List<Message> =
-      createPromptWithContextAwareOfTokens(
-        memories = memories,
-        ctxInfo = context.similaritySearch(prompt.message, promptConfiguration.docsInContext),
-        modelType = modelType,
-        prompt = prompt.message,
-        minResponseTokens = promptConfiguration.minResponseTokens
+    val messagesForRequest =
+      fitMessagesByTokens(
+        messagesFromMemory(memories),
+        prompt.toMessages(),
+        context,
+        modelType,
+        promptConfiguration
       )
 
-    val messages: List<Message> = messagesFromMemory(memories) + promptWithContext
-
-    fun checkTotalLeftChatTokens(): Int {
-      val maxContextLength: Int = modelType.maxContextLength
-      val messagesTokens: Int = tokensFromMessages(messages)
-      val totalLeftTokens: Int = maxContextLength - messagesTokens
-      if (totalLeftTokens < 0) {
-        throw AIError.MessagesExceedMaxTokenLength(messages, messagesTokens, maxContextLength)
-      }
-      return totalLeftTokens
-    }
-
-    val request: ChatCompletionRequest =
+    val request =
       ChatCompletionRequest(
         model = name,
         user = promptConfiguration.user,
-        messages = messages,
+        messages = messagesForRequest,
         n = promptConfiguration.numberOfPredictions,
         temperature = promptConfiguration.temperature,
-        maxTokens = checkTotalLeftChatTokens(),
+        maxTokens = promptConfiguration.minResponseTokens,
```

Comment:
I think that the field in the `PromptConfiguration` should be `maxResponseTokens` instead of `minResponseTokens`

Reply:
Let's address it in another PR for prompt improvements unless you want to change it here.

Reply:
In another PR is good to me